### PR TITLE
Issue 281: simplify stream reading

### PR DESCRIPTION
**Pull request by matozoid**
The second suggestion from: https://github.com/javaparser/javaparser/issues/281 and some small stuff

### CODE REVIEW

File: javaparser-core/src/main/java/com/github/javaparser/CommentsInserter.java [hunk 1]
```diff
@@ -21,32 +21,34 @@
...
+class CommentsInserter {
     private boolean doNotAssignCommentsPreceedingEmptyLines = true;
     private boolean doNotConsiderAnnotationsAsNodeStartForCodeAttribution = false;
+    
+    CommentsInserter() {
+    }
 
-    public void insertComments(CompilationUnit cu, String code)
-            throws IOException {
+    public void insertComments(CompilationUnit cu, String comments) throws IOException {
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/CommentsInserter.java]: It is not 100% clear to me what comments should contain: just the text of the comments? All the original code?

File: javaparser-core/src/main/java/com/github/javaparser/InstanceJavaParser.java [hunk 2]
```diff
@@ -80,7 +88,19 @@ public InstanceJavaParser(Reader reader) {
      *             if the source code has parser errors
      */
     public CompilationUnit parse() throws ParseException {
-        return astParser.CompilationUnit();
+        try {
+            return astParser.CompilationUnit();
+        }finally {
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/InstanceJavaParser.java]: Nitpicking, level master: we could add a space before finally

File: javaparser-core/src/main/java/com/github/javaparser/InstanceJavaParser.java [hunk 3]
```diff
@@ -91,7 +111,11 @@ public CompilationUnit parse() throws ParseException {
      *             if the source code has parser errors
      */
     public BlockStmt parseBlock() throws ParseException {
-        return astParser.Block();
+        try{
+            return astParser.Block();
+        }finally {
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/InstanceJavaParser.java]: Same here

File: javaparser-core/src/main/java/com/github/javaparser/InstanceJavaParser.java [hunk 4]
```diff
@@ -103,7 +127,11 @@ public BlockStmt parseBlock() throws ParseException {
      *             if the source code has parser errors
      */
     public List<?> parseStatements() throws ParseException {
-        return astParser.Statements();
+        try {
+            return astParser.Statements();
+        }finally {
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/InstanceJavaParser.java]: here

File: javaparser-core/src/main/java/com/github/javaparser/InstanceJavaParser.java [hunk 5]
```diff
@@ -115,7 +143,11 @@ public BlockStmt parseBlock() throws ParseException {
      *             if the source code has parser errors
      */
     public Statement parseStatement() throws ParseException {
-        return astParser.Statement();
+        try {
+            return astParser.Statement();
+        }finally {
```
Comment (ftomassetti) [inline javaparser-core/src/main/java/com/github/javaparser/InstanceJavaParser.java]: and there

**Comment by ftomassetti** [comment]
This seems a great patch! It is removing a lot of code. Good job!

File: javaparser-core/src/main/java/com/github/javaparser/CommentsInserter.java (hunk 1 above)
Comment (matozoid) [inline javaparser-core/src/main/java/com/github/javaparser/CommentsInserter.java]: I'm adding some javadoc here :)

**Pull request merged**